internal/configs/configschema/coerce_value.go

```diff
@@ -110,13 +110,15 @@ func (b *Block) coerceValue(in cty.Value, path cty.Path) (cty.Value, error) {
 					continue
 				}
 
+				coll, marks := coll.Unmark()
+
 				if !coll.CanIterateElements() {
 					return cty.UnknownVal(impliedType), path.NewErrorf("must be a list")
 				}
 				l := coll.LengthInt()
 
 				if l == 0 {
-					attrs[typeName] = cty.ListValEmpty(blockS.ImpliedType())
+					attrs[typeName] = cty.ListValEmpty(blockS.ImpliedType()).WithMarks(marks)
```

Comment:
Should this also be done for the case of `NestingSingle`? Or is it not possible for that block type to contain marks that get lost?

Also, does `convert.Convert` automatically retain the marks between the before & after?

Reply:
Sounds good. Thank you!